### DISCUSSION TITLE
[ENHANCEMENT]: ilogtail启动后会在可执行文件目录下生成一些文件，这些文件是否可以指定生成到当前工作目录或者特定目录？

### DISCUSSION THREAD

**Discussion by mmmyddd** [alibaba/ilogtail#1191] [Help] [answered]
ilogtail启动后会在可执行文件目录下生成一些文件，这些文件是否可以指定生成到当前工作目录或者特定目录？

**Comment by mmmyddd** [comment]
当前的使用场景是在容器中运行ilogtail，但是需要把ilogtail安装在容器外部，通过readonly形式bind mount 到多个容器内部，ilogtail可以在host上更新版本，如果这些临时文件总是生成在ilogtail安装目录，readonly mount会导致ilogtail安装目录无法执行

**Comment by yyuuttaaoo** [comment, marked as answer]
可以看一下这段代码，这些flag可以在ilogtail_config.json配置文件中改掉
``` C++
    STRING_FLAG(logtail_sys_conf_dir) = ".";
    STRING_FLAG(check_point_filename) = "checkpoint/logtail_check_point";
    STRING_FLAG(default_buffer_file_path) = "checkpoint";
    STRING_FLAG(ilogtail_docker_file_path_config) = "checkpoint/docker_path_config.json";
```
ilogtail_config.json可以修改后挂载挂进去
apsara_log_conf.json、plugin_logger.xml控制日志目录，也可以修改后挂载进去

**Comment by mmmyddd** [comment]
我试试，谢谢

**Reply by messixukejia** [reply]
https://github.com/alibaba/ilogtail/issues/693 麻烦登记下使用场景吧，多谢
